stock_storage_type/models/stock_location_storage_type.py

```diff
@@ -28,3 +29,68 @@ class StockLocationStorageType(models.Model):
         'stock_location_storage_type_id',
         'stock_package_storage_type_id',
     )
+
+    only_empty = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if there are not any existing quant nor planned move on "
+             "this location")
+    do_not_mix_lots = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if the location contains product of the same "
+             "lot."
+    )
+    do_not_mix_products = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if the location contains the same product."
+    )
+    # max_height = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging height is lower than this maximum."
+    # )
+    # max_height_value = fields.Float()
+    # max_volume = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging volume is lower than this maximum."
+    # )
+    # max_volume_value = fields.Float()
+
+    @api.constrains('only_empty', 'do_not_mix_lots', 'do_not_mix_products')
+    def _check_empty_mix(self):
+        for location_storage_type in self:
+            if (
+                location_storage_type.only_empty and (
+                    location_storage_type.do_not_mix_lots or
+                    location_storage_type.do_not_mix_products
+                )
+            ):
+                raise ValidationError(_(
+                    "You cannot set 'Do not mix lots' or 'Do not mix products'"
```

Comment:
you can not rise 2 different error to separate `Do not mix lots` and `Do not mix products` and may be add the lot name in the errror